membership-attribute-service/app/models/Attributes.scala

```diff
@@ -12,60 +10,87 @@ import play.api.mvc.Results.Ok
+                           MembershipJoinDate: Option[LocalDate] = None,
+                           DigitalSubscriptionExpiryDate: Option[LocalDate] = None)
+
+case class DynamoAttributes(UserId: String,
+                            Tier: Option[String] = None,
+                            RecurringContributionPaymentPlan: Option[String] = None,
+                            MembershipJoinDate: Option[LocalDate] = None,
+                            DigitalSubscriptionExpiryDate: Option[LocalDate] = None,
+                            MembershipNumber: Option[String],
+                            AdFree: Option[Boolean],
+                            TTLTimestamp: Long) {
+  lazy val isFriendTier = Tier.exists(_.equalsIgnoreCase("friend"))
+  lazy val isSupporterTier = Tier.exists(_.equalsIgnoreCase("supporter"))
+  lazy val isPartnerTier = Tier.exists(_.equalsIgnoreCase("partner"))
+  lazy val isPatronTier = Tier.exists(_.equalsIgnoreCase("patron"))
+  lazy val isStaffTier = Tier.exists(_.equalsIgnoreCase("staff"))
+  lazy val isPaidTier = isSupporterTier || isPartnerTier || isPatronTier || isStaffTier
+}
+
+object DynamoAttributes {
+  def asAttributes(dynamoAttributes: DynamoAttributes): Attributes = Attributes(
+    UserId = dynamoAttributes.UserId,
+    Tier = dynamoAttributes.Tier,
+    RecurringContributionPaymentPlan = dynamoAttributes.RecurringContributionPaymentPlan,
+    MembershipJoinDate = dynamoAttributes.MembershipJoinDate,
+    DigitalSubscriptionExpiryDate = dynamoAttributes.DigitalSubscriptionExpiryDate,
+    MembershipNumber = dynamoAttributes.MembershipNumber,
+    AdFree = dynamoAttributes.AdFree
+  )
+}
+
 object Attributes {
 
-  implicit val jsWrite: OWrites[Attributes] = (
+  implicit val jsAttributesWrites: OWrites[Attributes] = (
     (__ \ "userId").write[String] and
-    (__ \ "tier").writeNullable[String] and
-    (__ \ "membershipNumber").writeNullable[String] and
-    (__ \ "adFree").writeNullable[Boolean] and
-    (__ \ "recurringContributionPaymentPlan").writeNullable[String] and
-    (__ \ "membershipJoinDate").writeNullable[LocalDate] and
-    (__ \ "digitalSubscriptionExpiryDate").writeNullable[LocalDate] and
-      (__ \ "ttlTimestamp").writeNullable[Long]
-    )(unlift(Attributes.unapply)).addNullableField("digitalSubscriptionExpiryDate", _.latestDigitalSubscriptionExpiryDate).addField("contentAccess", _.contentAccess)
+      (__ \ "tier").writeNullable[String] and
+      (__ \ "recurringContributionPaymentPlan").writeNullable[String] and
+      (__ \ "membershipJoinDate").writeNullable[LocalDate] and
+      (__ \ "digitalSubscriptionExpiryDate").writeNullable[LocalDate] and
+      (__ \ "membershipNumber").writeNullable[String] and
+      (__ \ "adFree").writeNullable[Boolean]
+  )(unlift(Attributes.unapply)).addNullableField("digitalSubscriptionExpiryDate", _.latestDigitalSubscriptionExpiryDate).addField("contentAccess", _.contentAccess)
```

Comment:
existing code I know, but if you are in anyway and fancied splitting this onto a few physical lines it would help some one else